python/packages/autogen-core/src/autogen_core/_single_threaded_agent_runtime.py

```diff
@@ -532,6 +542,13 @@ async def stop(self) -> None:
         """Stop the runtime message processing loop."""
         if self._run_context is None:
             raise RuntimeError("Runtime is not started")
+
+        # deactivate all the agents that have been activated
+        for agent_id in self._activated_agents:
```

Comment:
Considering we just want shutdown, would be easier to just iterate `_instantiated_agents` here instead